example/azure-test-garden/README.md

```diff
@@ -0,0 +1,530 @@
+```
+
+## Deploy Gardener Helm Chart
+
+You need to update these Route53 related fields in the
+`example/azure-test-garden/gardener-values.yaml` since
+garden-controller-manager is the components that creates DNS records
+in Route53:
+* controller.internalDomain.hostedZoneID
+* controller.internalDomain.domain
+* controller.internalDomain.credentials
+* controller.internalDomain.secretAccessKey
+
+After updating these fields in the file, run:
+
+```
+helm upgrade --install --namespace garden garden charts/gardener -f example/azure-test-garden/gardener-values.yaml
+```
+
+Validate the gardener is deployed:
+
+```
+helm status garden  # Wait for `STATUS: DEPLOYED`
+
+kubectl -n garden get deploy,pod -l app=gardener
+
+# Better if you leave two terminals open in for below commands, and
+# keep an eye on whats going on behind the scenes as you create/delete
+# Gardener specific resources (Seed, CloudProfile, SharedSecret, Shoot).
+kubectl -n garden logs -f deployment/gardener-apiserver           # confirm no issues
+kubectl -n garden logs -f deployment/gardener-controller-manager  # confirm no issues
+```
+
+**Note**: This is not meant to be used in production. You may not want
+to use `apiserver.insecureSkipTLSVerify=true`, the hardcoded apiserver
+certs, and insecure (non-tls enabled) etcd. But for the sake of
+keeping this example simple you can just keep those values as they
+are.
+
+# Define Seed cluster in Gardener
+
+In our setup we'll use the cluster for Gardener also as a Seed, this
+saves us from creating a new Kubernetes cluster. But you can also
+create an explicit cluster for the Seed. Seed cluster can also be
+placed into any other cloud provider or on prem. But keep in mind that
+below steps may differ if you use a different cluster.
+
+## Create the Seed resource definition with its Secret
+
+Lets start with the required seed secret first. Here we need to
+provide it's cloud provider credentials and kubeconfig in the seed
+secret. Update `example/secret-seed-azure.yaml` and place the secrets
+for your environment:
+* **data.subscriptionID**: you can learn this one with `az account show`
+* **data.tenantID**: from `az ad sp create-for-rbac` output as you can see above
+* **data.clientID**: from `az ad sp create-for-rbac` output as you can see above
+* **data.clientSecret**: from `az ad sp create-for-rbac` output as you can see above
+* **data.kubeconfig**: you can get this one with `az aks get-credentials --resource-group garden-1 --name garden-1 -f - | base64`)
+
+**Note**: All of the above values must be base64 encoded. If you skip this it will hurt you later.
+
+After updating the fields, create the Seed secret:
+```
+kubectl apply -f example/secret-seed-azure.yaml
+```
+
+Before creating Seed, we need to update the `example/seed-azure.yaml` file and update
+* **spec.networks**: IP ranges used in your AKS cluster (TODO: how to get easily)
+* **spec.ingressDomain**: TODO: still not clear how is this exactly used
```

Comment:
This is used to generate ingress resources that are in the Seed cluster - for example every Shoot cluster gets a `Grafana`, `Prometheus` and `AlertManager` endpoints.